File: auction-backend/src/main/java/com/atlantbh/internship/auction/app/service/validator/user/LastNameValidator.java
package com.atlantbh.internship.auction.app.service.validator.user;

import com.atlantbh.internship.auction.app.service.validator.Validator;

public class LastNameValidator extends Validator {
    private static final Integer MIN_LAST_NAME_LENGTH = 3;
    private static final Integer MAX_LAST_NAME_LENGTH = 30;

    public static void validateLastName(final String input) {
        final String lastName = normaliseInput(input);

        isEmptyOrNull(lastName, "Last name field cannot be empty.");
        minimumLength(lastName, MIN_LAST_NAME_LENGTH, String.format("Last name cannot contain less than %d characters", MIN_LAST_NAME_LENGTH));
        maximumLength(lastName, MAX_LAST_NAME_LENGTH, String.format("Last name cannot contain more than %d characters", MAX_LAST_NAME_LENGTH));
        isMadeUpOfLetters(lastName, "The last name must consist exclusively of letters.");
    }
}

File: frontend/src/app/features/register/components/register-form/validators/last-name.validator.ts
import {NameValidator} from "./name-validator";
import NameValidatorError = NameValidator.NameValidatorError;

export class LastNameValidator {
  protected static readonly minLength: number = 3;
  protected static readonly maxLength: number = 30;

  public static message: Record<NameValidatorError, string> = {
    [NameValidatorError.Required]: `Last name field cannot be empty.`,
    [NameValidatorError.MinLength]: `Last name cannot contain less than ${this.minLength} characters.`,
    [NameValidatorError.MaxLength]: `Last name cannot contain more than ${this.maxLength} characters.`,
    [NameValidatorError.Letters]: `The last name must consist exclusively of letters.`,
  };

  public static config: NameValidator.Config = {
    minLength: LastNameValidator.minLength,
    maxLength: LastNameValidator.maxLength,
    messages: LastNameValidator.message
  };
}

